dev/PackageManager/API/M.W.M.D.PackageDeploymentManager.cpp

```diff
@@ -119,9 +141,14 @@ namespace winrt::Microsoft::Windows::Management::Deployment::implementation
         {
             case winrt::Microsoft::Windows::Management::Deployment::PackageDeploymentFeature::PackageUriScheme_ms_uup:
             {
-                //TODO Feature lookup
-                // Relies on PackageManagement_IsFeatureSupported(L"PackageUriScheme.ms-uup") exist in Microsoft.FrameworkUdk and enabled
-                return ::WindowsVersion::IsExportPresent(L"appxdeploymentclient.dll", "MsixRemovePackageByUriAsync");
+                BOOL isSupported{};
+                const HRESULT hr{ PackageManagement_IsFeatureSupported(L"PackageUriScheme.ms-uup", &isSupported) };
+                if (hr == E_NOTIMPL)
+                {
+                    return false;
+                }
+                THROW_IF_FAILED_MSG(hr, "PackageUriScheme_ms_uup");
+                return !!isSupported;
```

Comment:
Is the double ! a typo?

Reply:
Intentional

`BOOL` is a typedef for `USHORT` and `#define`s for `TRUE`=`1` and `FALSE`=`0`

`bool = BOOL` yields VS warning. The trick is to turn the `BOOL` integer expression to a boolean expression. `!BOOL` does that, but now the result is backwards -- `!TRUE`=`false` and `!FALSE`=`true`. Thus, `!!BOOL`=`bool`